R/checks.R

```diff
@@ -1,3 +1,43 @@
+# Additional type checkers designed for testing argument values.
+
+check_single_logical <- function(x, call = rlang::caller_env()) {
+ cl <- match.call()
+ arg_nm <- as.character(cl$x)
+ msg <- "{.arg {arg_nm}} should be a single logical value, not {brulee:::obj_type_friendly(x)}."
+ if (!is.logical(x)) {
+  cli::cli_abort(msg, call = call)
+ }
+ if (length(x) > 1 || any(is.na(x))) {
+  cli::cli_abort(msg, call = call)
+ }
+ invisible(x)
+}
+
+check_number_whole_vec <- function(x, call = rlang::caller_env(), ...) {
+ cl <- match.call()
+ arg <- as.character(cl$x)
+
+ for (i in x) {
```

Comment:
The downside here is that we only catch one violation at a time.

Reply:
I would imagine there's a pretty steep performance penalty here, too. This is the same kind of performance issue that led to https://github.com/tidymodels/parsnip/pull/835.

Maybe an error could be conditional on `rlang::is_integerish()`?

Reply:
Is it steep for checking arguments (not data)? We need some sort of (short length) numeric vector checks with min/max/na/null options.